### DISCUSSION TITLE
Send IDs of seen devices instead of counters

### DISCUSSION THREAD

**Discussion by sgaeumann** [cyberman54/ESP32-Paxcounter#955] [Q&A] [answered]
Hello. I uploaded the code in a TTGO T-Beam v1.1 and it works well. Now, I would like to adapt it to a specific use case where instead of sending the _pax_, _ble_ and _wifi_ counters on The Things Network, the device sends the IDs it has seen. I already took a look at the **libpax** library and at the **Paxcounter** code. 
This is how I think I would proceed : 

- Add a function in `libpax.cpp` that returns a pointer to the `seen_ids_map`
- Add a function in `libpax_api.cpp` that returns a pointer to the precedent pointer mentioned above
- In `main.cpp`, call the function created in `libpax_api.cpp`, go through the `seen_ids_map` and store the seen devices in a new array. This new array would only contain the seen IDs, and is therefore slightly different from the `seen_ids_map`.
- Modify the function `PayloadConvert::addCount()` in the `payload.cpp` file, to encode the array
- Modify the `senddata.cpp` file and adapt it to the modified `addCount()` function

1. Just for clarification : from what I understood by reading the **Libpax** library, the `seen_ids_map` is a sort of set that contains every possible ID (65535 possibilities). Whenever a device is added to the `seen_ids_map` (meaning it is the first time it is seen), it is signaled by changing a bit. Is it correct?
2. Is this procedure a good start, and is it feasible?

**Comment by cyberman54** [comment, marked as answer]
I won't support or merge such kind of feature, since it would be an unneeded privacy intrusion while the purpose is just to count pax, and it may trigger data protection legal issues, depending on country laws.

**Reply by sgaeumann** [reply]
Okay, thanks anyway.
